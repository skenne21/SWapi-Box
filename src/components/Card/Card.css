.card {
  display: inline-flex;
  flex-direction: column;
  width: 20%;
}

.planets {
  width: 80%;
  background-color: green;
}

.people {
  background-color: magenta;
}

.vehicle {
  background-color: purple;
}