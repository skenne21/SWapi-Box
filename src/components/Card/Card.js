import React from 'react';

const Card = (props) => {
  const planetPeepsCards = (props) => {
    if(props.climate) {
      
    }
  }

  const vechilesCards = (props) => {
    console.log(props)
  }

  return (
    <div>
    {
      props.population ? planetPeepsCards(props) : vechilesCards(props)
    }
    </div>
  )
}

export default Card;