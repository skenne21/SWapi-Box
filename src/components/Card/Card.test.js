import React from 'react';
import {shallow} from 'enzyme';
import Card from './Card.js';

describe('Card', () => {
  const card = {
    class: 'people',
    name: "Luke Skywalker",
    data: {
      species: "Human", 
      homeWorldName: "Tatooine", 
      population: "200000"
    }    
  }
  const mockToggleFavs = jest.fn();

  it('Should match the snapshot' , () => {
    const wrapper = shallow(
      <Card 
        card={card}
        className={'people'}
        toggleFavorites={mockToggleFavs}
      />)
    expect(wrapper).toMatchSnapshot();
  })
})