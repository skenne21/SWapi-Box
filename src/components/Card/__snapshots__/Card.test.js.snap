// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card={
        Object {
            "class": "people",
            "data": Object {
              "homeWorldName": "homeworld: Tatooine",
              "population": "Population: 200000",
              "species": "species: Human",
            },
            "name": "Luke Skywalker",
          }
    }
    className="people"
    toggleFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <article
        className="people"
>
        <h2>
                Luke Skywalker
        </h2>
        <Button
                card={
                        Object {
                                "class": "people",
                                "data": Object {
                                  "homeWorldName": "homeworld: Tatooine",
                                  "population": "Population: 200000",
                                  "species": "species: Human",
                                },
                                "name": "Luke Skywalker",
                              }
                }
                className="favs-card"
                controlFunc={[Function]}
                name="❤︎"
        />
        <h3
                className="detail"
        >
                species: Human
        </h3>
        <h3
                className="detail"
        >
                homeworld: Tatooine
        </h3>
        <h3
                className="detail"
        >
                Population: 200000
        </h3>
</article>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Luke Skywalker
</h2>,
          <Button
            card={
                        Object {
                                    "class": "people",
                                    "data": Object {
                                      "homeWorldName": "homeworld: Tatooine",
                                      "population": "Population: 200000",
                                      "species": "species: Human",
                                    },
                                    "name": "Luke Skywalker",
                                  }
            }
            className="favs-card"
            controlFunc={[Function]}
            name="❤︎"
/>,
          Array [
            <h3
              className="detail"
>
              species: Human
</h3>,
            <h3
              className="detail"
>
              homeworld: Tatooine
</h3>,
            <h3
              className="detail"
>
              Population: 200000
</h3>,
          ],
        ],
        "className": "people",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Luke Skywalker",
          },
          "ref": null,
          "rendered": "Luke Skywalker",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "card": Object {
              "class": "people",
              "data": Object {
                "homeWorldName": "homeworld: Tatooine",
                "population": "Population: 200000",
                "species": "species: Human",
              },
              "name": "Luke Skywalker",
            },
            "className": "favs-card",
            "controlFunc": [Function],
            "name": "❤︎",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "species: Human",
            "className": "detail",
          },
          "ref": null,
          "rendered": "species: Human",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "homeworld: Tatooine",
            "className": "detail",
          },
          "ref": null,
          "rendered": "homeworld: Tatooine",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "Population: 200000",
            "className": "detail",
          },
          "ref": null,
          "rendered": "Population: 200000",
          "type": "h3",
        },
      ],
      "type": "article",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <article
          className="people"
>
          <h2>
                    Luke Skywalker
          </h2>
          <Button
                    card={
                              Object {
                                        "class": "people",
                                        "data": Object {
                                          "homeWorldName": "homeworld: Tatooine",
                                          "population": "Population: 200000",
                                          "species": "species: Human",
                                        },
                                        "name": "Luke Skywalker",
                                      }
                    }
                    className="favs-card"
                    controlFunc={[Function]}
                    name="❤︎"
          />
          <h3
                    className="detail"
          >
                    species: Human
          </h3>
          <h3
                    className="detail"
          >
                    homeworld: Tatooine
          </h3>
          <h3
                    className="detail"
          >
                    Population: 200000
          </h3>
</article>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Luke Skywalker
</h2>,
            <Button
              card={
                            Object {
                                          "class": "people",
                                          "data": Object {
                                            "homeWorldName": "homeworld: Tatooine",
                                            "population": "Population: 200000",
                                            "species": "species: Human",
                                          },
                                          "name": "Luke Skywalker",
                                        }
              }
              className="favs-card"
              controlFunc={[Function]}
              name="❤︎"
/>,
            Array [
              <h3
                className="detail"
>
                species: Human
</h3>,
              <h3
                className="detail"
>
                homeworld: Tatooine
</h3>,
              <h3
                className="detail"
>
                Population: 200000
</h3>,
            ],
          ],
          "className": "people",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Luke Skywalker",
            },
            "ref": null,
            "rendered": "Luke Skywalker",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "card": Object {
                "class": "people",
                "data": Object {
                  "homeWorldName": "homeworld: Tatooine",
                  "population": "Population: 200000",
                  "species": "species: Human",
                },
                "name": "Luke Skywalker",
              },
              "className": "favs-card",
              "controlFunc": [Function],
              "name": "❤︎",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "species: Human",
              "className": "detail",
            },
            "ref": null,
            "rendered": "species: Human",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "homeworld: Tatooine",
              "className": "detail",
            },
            "ref": null,
            "rendered": "homeworld: Tatooine",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "Population: 200000",
              "className": "detail",
            },
            "ref": null,
            "rendered": "Population: 200000",
            "type": "h3",
          },
        ],
        "type": "article",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
